### DISCUSSION TITLE
Transposing capability for not standard MIDI

### DISCUSSION THREAD

**Discussion by cruedamerch** [GrandOrgue/grandorgue#1042] [Help] [answered]
Hello,
I am trying to map a not standard midi keyboard, but the keyboard is not sending out standard midi messages and, once mapped, C4 on the keyboard is C1 in GO...  I tried to use the Transpose function in the keyboard mapping dialog, but 35 semitones is the maximum it allows...  36 would be perfect to solve my issue (3 octaves up)...  ¿is there any other option in GO I could solve this problem I have?  It is impossible to change the midi out information the keyboard is sending...

Thank you very much!!

**Comment by midimusic** [comment]
Bear in mind that the only allowable midi notes run from 0 to 127. How does your keyboard cope with this?

Use Midi Ox or a similar program to find out what your keyboard REALLY does. I suspect it is faulty and in any case you would be better off with one which works normally.

csw900

**Comment by cruedamerch** [comment]
Thank you for your response!  I checked what midi notes are being sent, and the keyboard is distributed not as the standard ones in which C3 is key 60 in terms of midi messages...  C1 on the keyboard which has 5 octaves, is key 0...  therefore everything would work transposing 3 octaves up...  it is a real pipe organ which has been digitalized, and the manufacturer electronics cannot be modified...  but I would like to add the virtuality G.O. allows...

The problem is that in G.O. the max semitones I am able to transpose (from the keyboard mapping window) is 35 semitones... this is one semitone less to accomplish 3 octaves...   So I am wondering if there is any other workaround to transpose everything 3 octaves up...

Thank you!

Obtener Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: midimusic ***@***.***>
Enviado: Sunday, February 20, 2022 9:36:36 AM
Para: GrandOrgue/grandorgue ***@***.***>
Cc: cruedamerch ***@***.***>; Author ***@***.***>
Asunto: Re: [GrandOrgue/grandorgue] Transposing capability for not standard MIDI (Discussion #1042)


Bear in mind that the only allowable midi notes run from 0 to 127. How does your keyboard cope with this?

Use Midi Ox or a similar program to find out what your keyboard REALLY does. I suspect it is faulty and in any case you would be better off with one which works normally.

csw900

—
Reply to this email directly, view it on GitHub<https://github.com/GrandOrgue/grandorgue/discussions/1042#discussioncomment-2213523>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AX22AYOS73VQK3GZKQAFSJTU4CRZJANCNFSM5OYML75Q>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by midimusic** [comment, marked as answer]
You could try using the keyboard mapping to transpose 35 semitones and get the other semitone from the main panel transpose (which can do -11 to +11). It surprises me that GO even allows 35 semitones on the keyboard mapper - this is very generous.

Other than suggesting you modify the GO code and make a special version of GO for your non standard organ I do not think I can help any more. What you really need is a little program to interpose between you keyboard and GO which does the transposing and feeds standard midi to GO. This would be very easy for an experienced programmer  to write

csw900

**Reply by cruedamerch** [reply]
I tried the first option you suggested.... and the issue was solved!  thanks, great help

**Comment by cruedamerch** [comment]
I will try the first option you’re suggesting and check if it works... it sounds good to me... I’ll let you know the result...

Thanx!

Obtener Outlook para iOS<https://aka.ms/o0ukef>
________________________________
De: midimusic ***@***.***>
Enviado: Friday, February 25, 2022 12:57:00 PM
Para: GrandOrgue/grandorgue ***@***.***>
Cc: cruedamerch ***@***.***>; Author ***@***.***>
Asunto: Re: [GrandOrgue/grandorgue] Transposing capability for not standard MIDI (Discussion #1042)


You could try using the keyboard mapping to transpose 35 semitones and get the other semitone from the main panel transpose (which can do -11 to +11). It surprises me that GO even allows 35 semitones on the keyboard mapper - this is very generous.

Other than suggesting you modify the GO code and make a special version of GO for your non standard organ I do not think I can help any more. What you really need is a little program to interpose between you keyboard and GO which does the transposing and feeds standard midi to GO. This would be very easy for an experienced programmer to write

csw900

—
Reply to this email directly, view it on GitHub<https://github.com/GrandOrgue/grandorgue/discussions/1042#discussioncomment-2251053>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AX22AYNLDG2TZHBI7MYJ433U45VAZANCNFSM5OYML75Q>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
